src/hotspot/share/oops/constantPool.hpp

```diff
@@ -77,6 +77,36 @@ class CPKlassSlot {
   }
 };
 
+class BSMAttributeEntry {
+  friend class ConstantPool;
+
+  u2 _bootstrap_method_index;
+  u2 _argument_count;
+
+  const u2* argument_indexes() const {
+    return reinterpret_cast<const u2*>(this + 1);
+  }
+  u2* argument_indexes() {
+    return reinterpret_cast<u2*>(this + 1);
+  }
+
+  // These are overlays on top of the operands array. Do not construct.
+  BSMAttributeEntry() = delete;
+
+public:
+  int bootstrap_method_index() const {
+    return _bootstrap_method_index;
+  }
+  int argument_count() const {
+    return _argument_count;
+  }
+
+  int argument_index(int n) const {
+    assert(checked_cast<u2>(n) < _argument_count, "oob");
```

Comment:
Should the assert contain a check that _argument_count is >= 0 as well?

Reply:
I don't think so, it's a `u2` so that's just part of its type.